content/en/blog/2024/gateway-mesh-ga/index.md

```diff
@@ -0,0 +1,98 @@
+
+Throughout the Istio documentation, all of our examples have been updated to show how to use the Gateway API, so explore some of the [tasks](/docs/tasks/traffic-management/) to gain a deeper understanding.
+
+Using Gateway API for service mesh should feel familiar both to users already using Gateway API for ingress, and users using `VirtualService` for service mesh today.
+
+* Compared to Gateway API for ingress, routes target a `Service` instead of a `Gateway`.
+* Compared to `VirtualService`, where routes associate with a set of `hosts`, routes target a `Service`.
+
+Here is a simple example, which demonstrates routing requests to two different versions of a `Service` based on the request header:
+
+{{< text yaml >}}
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: reviews
+spec:
+  parentRefs:
+  - group: ""
+    kind: Service
+    name: reviews
+    port: 9080
+  rules:
+  - matches:
+    - headers:
+      - name: my-favorite-service-mesh
+        value: istio
+    filters:
+    - type: RequestHeaderModifier
+      requestHeaderModifier:
+      add:
+        - name: hello
+          value: world
+    backendRefs:
+    - name: reviews-v2
+      port: 9080
+  - backendRefs:
+    - name: reviews-v1
+      port: 9080
+{{< /text >}}
+
+Breaking this down, we have a few parts:
+* First, we identify what routes we should match.
+  By attaching our route to the `reviews` Service, we will apply this routing configuration to all requests that were originally targeting `reviews`.
+* Next, `matches` configures criteria for selecting which traffic this route should handle.
+* Optionally, we can modify the request. Here, we add a header.
+* Finally, we select a destination for the request. In this example, we are picking between two versions of our application.
+
+For more details, see [Istio's traffic routing internals](/docs/ops/configuration/traffic-management/traffic-routing/) and [Gateway API's Service documentation](https://gateway-api.sigs.k8s.io/mesh/service-facets/).
+
+## Which API should I use?
+
+With overlapping responsibilities (and names!), picking which APIs to use can be a bit confusing.
+
+Here is the breakdown:
+
+| API Name     | Object Types                                                                                                                          | Status                            | Recommendation                                                             |
+|--------------|---------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------|----------------------------------------------------------------------------|
+| Gateway APIs | [HTTPRoute](https://gateway-api.sigs.k8s.io/api-types/httproute/), [Gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/), ... | Stable in Gateway API v1.0 (2023) | Use for new deployments, in particular with [ambient mode](/docs/ambient/) |
```

Comment:
Is Gateway API for sidecar stable but for ambient remains beta as ambient is still beta?

Reply:
The stability of **any** feature is the `min(intersection of features used)`. For exampe, I think we claim multi-clsuter in istio is beta (whether it should be is out of scope for this comment :slightly_smiling_face: ).

We don't need to say "Gateway API for Multiclsuter is beta" and "Gateway API for dual stack is experimental" and "Gateway API for ambient is beta".

Reply:
Understood, just thought users would ask that question as they read the blog as ambient mode is mentioned here.